lib/SqlConnection.py

```diff
@@ -0,0 +1,149 @@
+import os,sys
+import logging
+import logging.handlers
+from cloghandler import ConcurrentRotatingFileHandler
+from collections import deque
+from sqlalchemy import *
+from sqlalchemy.dialects.postgresql import ARRAY
+
+class PipelineSqlConnection:
+
+    def __init__(self,**kwargs):
+        self.logger = kwargs.get('logger',None)
+        if not self.logger:
+            self.initializeLogging(**kwargs)
+
+        self.host = kwargs.get('HOST','localhost')
+        self.port = kwargs.get('PORT','5432')
+        self.database = kwargs.get('DATABASE',None)
+        self.user = kwargs.get('USER','SpacemanSteve')
+        self.password = kwargs.get('PASSWD',None)
+        self.schema = kwargs.get('SCHEMA','nonbib')
+        self.meta = MetaData()
+
+        auth = ''
+        if self.user and self.password:
+            auth =  '%s@' % (':'.join([self.user,self.password]))
+        elif self.user:
+            auth = self.user + '@'
+        self.uri = 'postgres://%s%s:%s/%s' % (auth,self.host,self.port,self.database)
+        self.logger.info('solr url: {}, schema: {}'.format(self.uri, self.schema))
+
+
+        #self.engine = create_engine('postgresql://SpacemanSteve@localhost:5432/unittests_', echo=False)
+        self.engine = create_engine(self.uri, echo=False)
+        self.conn = self.engine.connect()
+
+        # sql row view table
+    def get_row_view_table(self, meta=None):
+        if meta is None:
+            meta = self.meta
+        return Table('rowviewm', meta,
+                     Column('bibcode', String, primary_key=True),
+                     Column('id', Integer),
+                     Column('authors', ARRAY(String)),
+                     Column('refereed', Boolean),
+                     Column('simbad_objects', ARRAY(String)),
+                     Column('grants', ARRAY(String)),
+                     Column('citations', ARRAY(String)),
+                     Column('boost', Float),
+                     Column('citation_count', Integer),
+                     Column('read_count', Integer),
+                     Column('norm_cites', Integer),
+                     Column('readers', ARRAY(String)),
+                     Column('downloads', ARRAY(Integer)),
+                     Column('reads', ARRAY(Integer)),
+                     Column('reference', ARRAY(String)),
+                     schema=self.schema,
+                     extend_existing=True)
+
+
+    def getAllBibcodes(self):
+        return_value = deque()
+        row_view = self.get_row_view_table()
+        s = select([row_view])
+        rp = self.conn.execute(s)
+        rows = rp.fetchall()
```

Comment:
worst case is 12m, I added a docstring saying it is only called by test code to verify the sql table was initialized with test data.